src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -366,6 +366,21 @@ private void CompileMethodInternal(IMethodNode methodCodeNodeNeedingCode, Method
 #endif
             }
 
+            if (codeSize < _code.Length)
+            {
+                if (_compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.X64 ||
+                    _compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.X86)
+                {
+                    // For xarch, the generated code is sometimes smaller than the memory allocated.
```

Comment:
Is the code always of the right size outside xarch?

If yes, should the if-check above be assert instead?

Reply:
There may be cases of `codeSize != _code.Length`. We just return the native code size through `codeSize` out parameter. There is no way to know how much of the `_code.Length` was part of ro data, so there could be a mismatch.

https://github.com/dotnet/runtime/blob/c580f49f6527ac8df28876c4156345bc0a73ba64/src/coreclr/jit/emit.cpp#L5875-L5895